libs/dyn/dynvar/ref.go

```diff
@@ -1,12 +1,16 @@
 package dynvar
 
 import (
+	"fmt"
 	"regexp"
 
 	"github.com/databricks/cli/libs/dyn"
 )
 
-var re = regexp.MustCompile(`\$\{([a-zA-Z]+([-_]?[a-zA-Z0-9]+)*(\.[a-zA-Z]+([-_]?[a-zA-Z0-9]+)*(\[[0-9]+\])*)*(\[[0-9]+\])*)\}`)
+var (
+	baseVarDef = `[a-zA-Z]+([-_]*[a-zA-Z0-9]+)*`
+	re         = regexp.MustCompile(fmt.Sprintf(`\$\{(%s(\.%s(\[[0-9]+\])*)*(\[[0-9]+\])*)\}`, baseVarDef, baseVarDef))
```

Comment:
Can you check if the second `(\[[0-9]+\])*` is actually needed?

It looks like a repetition of the previous pattern.

Not blocking.

Reply:
Also, the pattern requires you to not have a `-` or `_` at the end.

We can make this more broad and use `[a-zA-Z]+[-_a-zA-Z0-9]*`.